src/CanvasPattern.cc

```diff
@@ -73,8 +75,10 @@ NAN_METHOD(Pattern::New) {
  * Initialize linear gradient.
  */
 
-Pattern::Pattern(cairo_surface_t *surface) {
+Pattern::Pattern(cairo_surface_t *surface, repeat_type_t *repeat) {
   _pattern = cairo_pattern_create_for_surface(surface);
+  _repeat = repeat;
```

Comment:
(continued) the 2nd arg would be `repeat_type_t repeat`, and then this constructor would set the pointer value.